homeworks/Andrii.Morozov_AndriiMorozov88/10-components/tik-tak-toe/src/app/field/field.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, OnInit } from '@angular/core';
+import { DataService } from '../data.service';
+import { RxjsService } from '../rxjs.service';
+import { Player } from '../enums';
+@Component({
+    selector: 'app-field',
+    templateUrl: './field.component.html',
+    styleUrls: ['./field.component.css']
+})
+export class FieldComponent implements OnInit {
+    constructor(public dataService: DataService, private rxjsService: RxjsService) {
+        this.rxjsService.currentPlayer$.subscribe(player => this.dataService.player = player);
+    }
+    ngOnInit(): void {
+        this.dataService.currentReset();
+    }
+    move(index:number):void {
+        if (this.dataService.fieldsquares[index] === Player.empty && this.dataService.gameOver === false) {
+            if (this.dataService.player === Player.cross) {
+                this.dataService.fieldsquares[index] = Player.cross;
+            } else {
+                this.dataService.fieldsquares[index] = Player.zero;
+            }
+            this.dataService.checkWinCombination();
+            this.dataService.togglePlayer();
+            this.rxjsService.changePlayer(this.dataService.player);
+        }
+    }
```

Comment:
Старайся робити менше вкладеностей

```suggestion
        if (this.dataService.fieldsquares[index] !== Player.empty) return;
        if (this.dataService.gameOver) return;
        
        if (this.dataService.player === Player.cross) {
            this.dataService.fieldsquares[index] = Player.cross;
        } else {
            this.dataService.fieldsquares[index] = Player.zero;
        }
        this.dataService.checkWinCombination();
        this.dataService.togglePlayer();
        this.rxjsService.changePlayer(this.dataService.player);
    }
```